### PR TITLE
PE-1102: 2022-10-19 Executive

### PR DESCRIPTION
**Pull request by iamchrissmith**
# Description

# Contribution Checklist

- [ ] PR title starts with `(PE-<TICKET_NUMBER>)`
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [ ] Every contract variable/method declared as public/external private/internal
- [ ] Consider if this PR needs the `officeHours` modifier override
- [ ] Verify expiration (`30 days` unless otherwise specified)
- [ ] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [ ] Validate all addresses used are in Goerli changelog or known
- [ ] Notify any external teams affected by the spell so they have the opportunity to review
- [ ] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [ ] Ensure contract is verified on `Goerli` etherscan
- [ ] Change test to use Goerli spell address and deploy timestamp
- [ ] Cast spell on Goerli `make spell="0x-deployed-spell-address" cast-spell`
- [ ] Run `make archive-spell` or `make date="YYYY-MM-DD" archive-spell` to make an archive directory and copy `Goerli-DssSpell.sol`, `Goerli-DssSpell.t.sol`, `Goerli-DssSpell.t.base.sol`, and `Goerli-DssSpellCollateralOnboarding.sol`
- [ ] `squash and merge` this PR

### CODE REVIEW

File: src/Goerli-DssSpell.sol
Comment (godsflaw) [inline src/Goerli-DssSpell.sol:48]: nit

```suggestion
    uint256 internal constant WAD = 10 ** 18;
```

**Review by godsflaw — APPROVED** [review]
LGTM, but I left one nit

```haskell
Running 1 test for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 733706)
Test result: ok. 1 passed; 0 failed; finished in 217.49ms

Running 15 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 506950)
[PASS] testFailTooEarly() (gas: 12594)
[PASS] testFailTooLate() (gas: 12549)
[PASS] testFailWrongDay() (gas: 12594)
[PASS] testFail_notScheduled() (gas: 14161)
[PASS] testOnTime() (gas: 503361)
[PASS] testSpellIsCast_GENERAL() (gas: 22851255)
[PASS] testSpellIsCast_PSM_GUSD_A_tout() (gas: 513507)
[PASS] test_auth() (gas: 21010735)
[PASS] test_auth_in_sources() (gas: 4946400)
[PASS] test_bytecode_matches() (gas: 1157265)
[PASS] test_chainlog_values() (gas: 5807739)
[PASS] test_chainlog_version_bump() (gas: 3045990)
[PASS] test_nextCastTime() (gas: 351611)
[PASS] test_use_eta() (gas: 350466)
Test result: ok. 15 passed; 0 failed; finished in 7.00s
```

**Review by amusingaxl — DISMISSED** [review]
LGTM :shipit:

File: src/Goerli-DssSpellCollateral.sol
Comment (The-Arbiter) [inline src/Goerli-DssSpellCollateral.sol:35]: Do we want this in this file or Goerli-DssSpell.sol?

File: src/Goerli-DssSpellCollateral.sol
Comment (The-Arbiter) [inline src/Goerli-DssSpellCollateral.sol:40]: Config uses 604_800 (no unit keyword) with 'weeks', proposing we match this.

```suggestion
    uint256 constant RWA007_A_AUTOLINE_TTL    = 604_800; // 1 week 
```

File: src/Goerli-DssSpell.sol
Comment (The-Arbiter) [inline src/Goerli-DssSpell.sol:67]: Any reason we don't do it like this for readability?

```suggestion
        DssExecLib.setValue(MCD_PSM_GUSD_A, "tout", 20 * WAD / 100_00);
```

Up to you, especially wrt using 100_00 instead of 10000.

File: src/test/config.sol
Comment (The-Arbiter) [inline src/test/config.sol:1378]: Point on this in my checklist does setIlk() on the IAM change this?

File: src/test/config.sol
Comment (The-Arbiter) [inline src/test/config.sol:1374]: Is this being changed by setIlkAutoLineParameters() (which calls setIlk() )?

**Review by The-Arbiter — DISMISSED** [review]
# Goerli Executive Spell Review Checklist (PE-1102)

PR: https://github.com/makerdao/spells-goerli/pull/119

Spell Crafter: CS
Reviewers: Arby, Mooney

Spell Actions:

- Set RWA007 IAM line to 250m
- Set RWA007 IAM gap to 50m
- Set RWA007 IAM ttl (cooldown) to 604,800 seconds
- Set GUSD PSM tout (fee out) to 20 bps == 0.2%
- Set GUSD PSM line to 500m
- Set GUSD PSM gap to 50m

## Checklist
- [x] Office Hours 
    - [x] off (default, spell casted after deploying, pause delay 60 sec.)
- [x] Exec Hash 
    - [x] empty (default)
- [x] 30 Days Expiry
    - [x] weekly (monthly discarded)
- [x] Pragma OK
- [x] Math OK
    - [x] Internal Precision
        - [x] `WAD = 10**18`
    - [x] Units
        - [x] ✅ `MILLION = 10**6` <--- Previous pattern [here](https://github.com/makerdao/spells-goerli/blob/PE-1064/src/Goerli-DssSpell.sol) was to describe MILLION in the DssSpell.sol. I know it's in DssSpellCollateral but do we want to / can we move it here next to WAD?
        * [x] ✅ Time: `RWA007_A_AUTOLINE_TTL = 1 weeks;` - do we want this as seconds or weeks? Previous spell PE-1098 opted not to use `seconds` keyword. Nitpicking here, the choice is yours, non-blocking issue. Config.sol uses 604_800 but the spell uses `weeks` - should we make this consistent?
* [x] DSS Deployed Contracts (e.g. Clips, Calcs, ...)
    GUSD PSM address is OK
    RWA007 address derived through dss exec lib => commit hash `c70218a` matches previously used commit hashes, address is assumed to be OK. Chainlog returned value looks OK. Exec lib address is OK. 

* [x] Collateral Onboarding
	RWA007 changes
    * [x] Autoline ([setIlkAutoLineParameters](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L648))
        * [x] `ilk` => Matches chainlog
        * [x] `line` => Units correct, matches doc
        * [x] `gap` => Units correct, matches doc
        * [x] `ttl` => Correct, I raised concern about the 'weeks' keyword above
        
	GUSD PSM changes
    * [x] Autoline ([setIlkAutoLineParameters](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L648))
        * [x] `ilk` => Matches chainlog
        * [x] `line` => Units correct, matches doc
        * [x] `gap` => Units correct, matches doc
        * [x] `ttl` => Existing TTL is 86400, which matches 24 hours provided as input, OK.
              
        ✅ Non-issue (comment): Do we want to note that we need a keeper to interact permissionlessly for these changes to be implemented? We did this for the PE-1098 mainnet spell.
* [x] RWA Updates
* [x] `addresses_goerli.sol` Matches
      RWA007 and GUSD PSM match
* [x] Spell Actions Look Good
	GUSD Tout change => Precision should be WAD per the GUSD-PSM contract. Matches precision used. 
	🛑 I propose that we use 20 * WAD / 100_00 to be clearer and more readable. Up to you.
* [x] Tests Look Good
      Checks RWA007 values are correct => OK
      Checks GUSD PSM values are correct => ✅ Relies on 'hours' => nitpicking but RWA doesn't use `weeks` and GUSD uses `hours`. It's used a bunch elsewhere so just pointing this out for consistency.
      Checks that tout changes to 0.2% => OK 
      Turns off 'line' for RWA007 => going to rabbithole into if `setIlkAutoLineParameters` sets line to 0 or not:
      *SetIlk() is called on DCIAM by DssExecLib, which [assigns `ilks[ilk]` to the input AutoLine profile](https://github.com/makerdao/dss-auto-line/blob/master/src/DssAutoLine.sol#L81)*
      ✅ Can you please confirm that SetIlk actually changes the existing `line` from 1M to 0? All I can see is ilks[ilk] being assigned and Setup being emitted but maybe I'm missing something. Edit: this was an internally used testing variable.

File: src/Goerli-DssSpellCollateral.sol
Comment (iamchrissmith) [reply src/Goerli-DssSpellCollateral.sol:35]: It has to be in this file since they use it here.

File: src/Goerli-DssSpellCollateral.sol
Comment (iamchrissmith) [reply src/Goerli-DssSpellCollateral.sol:40]: I would prefer to use two different measures since config is a test and therefore provides a check that 1 weeks is what we expect and that 604_800 seconds is one week

File: src/Goerli-DssSpell.sol
Comment (iamchrissmith) [reply src/Goerli-DssSpell.sol:67]: I took this pattern from a different spell.  My gut says not to add a 0 to both

File: src/Goerli-DssSpellCollateral.sol
Comment (godsflaw) [reply src/Goerli-DssSpellCollateral.sol:40]: So, we test this in terms of seconds later in the test.  It think I prefer this notation for now, so long as we are testing against the seconds value in the integration tests (which we are).

File: src/test/config.sol
Comment (godsflaw) [reply src/test/config.sol:1378]: I think it's more that the test no longer validates line once autoline is on (since autoline adjusts this).

File: src/Goerli-DssSpell.sol
Comment (The-Arbiter) [reply src/Goerli-DssSpell.sol:67]: There is no overflow risk I am seeing (evaluation of the value is determined only by constants) and it is more readable with 20 * WAD (representing 20bps), but I'll leave this up to you as it is non-blocking.

File: src/test/config.sol
Comment (The-Arbiter) [reply src/test/config.sol:1378]: You're correct, this is an internal bool used by DssSpell.t.base to determine what sets of config values to check.

Either way IMO I would leave line as 1M since that is the actual value unless we are changing this - is there a previous pattern for this? 

```suggestion
            line:         1_000_000,
```

File: src/Goerli-DssSpellCollateral.sol
Comment (The-Arbiter) [reply src/Goerli-DssSpellCollateral.sol:40]: Makes sense, no issue here then.

**Review by The-Arbiter — APPROVED** [review]
Test status: TODO

*Approving to not hold up deployment, will run later.*

**Review by godsflaw — APPROVED** [review]
ready for deploy

```haskell
[PASS] testStarknet() (gas: 733706)
Test result: ok. 1 passed; 0 failed; finished in 392.29ms

Running 15 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 506950)
[PASS] testFailTooEarly() (gas: 12594)
[PASS] testFailTooLate() (gas: 12549)
[PASS] testFailWrongDay() (gas: 12594)
[PASS] testFail_notScheduled() (gas: 14161)
[PASS] testOnTime() (gas: 503361)
[PASS] testSpellIsCast_GENERAL() (gas: 22851255)
[PASS] testSpellIsCast_PSM_GUSD_A_tout() (gas: 513507)
[PASS] test_auth() (gas: 21010735)
[PASS] test_auth_in_sources() (gas: 4946400)
[PASS] test_bytecode_matches() (gas: 1157265)
[PASS] test_chainlog_values() (gas: 5807739)
[PASS] test_chainlog_version_bump() (gas: 3045990)
[PASS] test_nextCastTime() (gas: 351611)
[PASS] test_use_eta() (gas: 350466)
Test result: ok. 15 passed; 0 failed; finished in 10.90s
```

**Comment by iamchrissmith** [comment]
deployed:
https://goerli.etherscan.io/tx/0x183011ee37d86833ec9bc9fc7834892be7e3d775eb4556ddc87fec93a7b90089
https://goerli.etherscan.io/address/0x23990e11a1efd07a6000af0e827a47f2f2086b6e#code

```rust
Running 1 test for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 733706)
Test result: ok. 1 passed; 0 failed; finished in 80.92ms

Running 15 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 506950)
[PASS] testFailTooEarly() (gas: 12594)
[PASS] testFailTooLate() (gas: 12549)
[PASS] testFailWrongDay() (gas: 12594)
[PASS] testFail_notScheduled() (gas: 14161)
[PASS] testOnTime() (gas: 503361)
[PASS] testSpellIsCast_GENERAL() (gas: 23996893)
[PASS] testSpellIsCast_PSM_GUSD_A_tout() (gas: 513507)
[PASS] test_auth() (gas: 21010735)
[PASS] test_auth_in_sources() (gas: 4946400)
[PASS] test_bytecode_matches() (gas: 1157265)
[PASS] test_chainlog_values() (gas: 5807739)
[PASS] test_chainlog_version_bump() (gas: 3045990)
[PASS] test_nextCastTime() (gas: 351611)
[PASS] test_use_eta() (gas: 350466)
Test result: ok. 15 passed; 0 failed; finished in 776.34ms
```

ready for archive approval

**Review by godsflaw — DISMISSED** [review]
LGTM, ready for archive

```haskell
Running 1 test for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 733706)
Test result: ok. 1 passed; 0 failed; finished in 220.14ms

Running 15 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 506950)
[PASS] testFailTooEarly() (gas: 12594)
[PASS] testFailTooLate() (gas: 12549)
[PASS] testFailWrongDay() (gas: 12594)
[PASS] testFail_notScheduled() (gas: 14161)
[PASS] testOnTime() (gas: 503361)
[PASS] testSpellIsCast_GENERAL() (gas: 23996893)
[PASS] testSpellIsCast_PSM_GUSD_A_tout() (gas: 513507)
[PASS] test_auth() (gas: 21010735)
[PASS] test_auth_in_sources() (gas: 4946400)
[PASS] test_bytecode_matches() (gas: 1157265)
[PASS] test_chainlog_values() (gas: 5807739)
[PASS] test_chainlog_version_bump() (gas: 3045990)
[PASS] test_nextCastTime() (gas: 351611)
[PASS] test_use_eta() (gas: 350466)
Test result: ok. 15 passed; 0 failed; finished in 6.92s
```

**Review by The-Arbiter — DISMISSED** [review]
Tests pass, ready to archive

**Review by godsflaw — APPROVED** [review]
archive looks good, tests still pass, ready for merge and cast.

**Review by The-Arbiter — APPROVED** [review]
lgtm-erge and cast

**Pull request merged**
